Msys bash

daxgames opened the pull request:
Added support for MSys2 Bash.
Renamed cmd task to cmder.
Also added personal files to .gitignore so they never get uploaded to the repo.

Stanzilla commented:
Those are way too many unrelated changes for one PR, imo.

daxgames commented:
The development branch had no support for a bash shell as previous version did, as for the  changes and why I saw _most_ of them as related.  Note: See below for possible unrelated changes that could be backed out.

1. Adding 'vendor/cmder.sh' - Is required for Msys2 bash, to normalize $CMDER_ROOT for the unix like env and to get git into the path in a portable fashion.
2. Modifying 'scripts\build.ps1' - Is required for Msys2 bash, to place 'cmder.sh' in /etc/profile.d post download and expansion of vendor\git-for-windows
3. Modifying 'packignore' - Is required for Msys2 bash, to ignore 'vendor/cmder.sh' during packaging since it does not need to be there in the final packaged distribution because it was put in '.\vendor\git-for-windows\etc\profile.d\cmder.sh' buy the changed 'scripts\build.ps1'
4. Modifying '.\config\ConEmu.xml' - Is Required for Msys2 Bash, to add the task to launch this shell.

I did the four above changes in the same pull request for the sake of completeness and to match what I saw was happening with the Cmd.exe and Powershell.exe based tasks already defined in the 'development' branch.

If you prefer I can back out the two seemingly unrelated changes I mentioned below and resubmit but in my opinion the .gitignore changes are required and should stay.

Possible unrelated changes:

1. '.gitignore' changes - This just looked like an oversight to me that needed to me fixed since the files are auto created on first run of each shell type and you don't want them in the repo.
2. Renaming task 'cmd' to 'cmder' in 'config\conemu.xml' - This was in the same file that required changes to add a task msys2 bash in change 4 above.   Most of the changes in here are time/datestamps written by ConEmu that I had no control of that will not affect the operation of ConEmu.

MartiUK commented:
Why are you replacing cmd with bash? Is this something you think everyone is going to want?

daxgames commented:
I was not replacing cmd with bash, at least I don't think I did.  I was adding bash using mintty.exe as an additional option because there was none in the 'development' branch I pulled down and I could not figure out the ConEmu commands required to get it woking with cmd.  

There were three tasks cmd, powershell, and powershell as admin there was no bash.  I added the 4th task.

Maybe someone added the 4th task launching bash from cmd between when I pulled the development branch and when I created the pull request?  I did not even notice it was a change vs. an add in the pull request until now.

daxgames commented:
Actually I just pullde down a clean copy of development and there are still only 3 options.  No bash.

JanSchulz commented on vendor/cmder.sh:
```diff
@@ -0,0 +1,29 @@
+# DO NOT EDIT FIS FILE IT WILL BE OVERWRITTEN ON UPDATE
```
s/FIS/THIS/

Stanzilla commented:
Ok from how I understood it, you wanted to add a new task for bash, but you renamed and changed the default one from cmd to bash?

daxgames commented:
No, I was trying to add a 4th task because there was no bash task when I look at con-emu tasks.  so I added a 4th task.  Since the task count in conemu.xml was 3 instead of 4 my attempted 'adding' of a 4th task actually 'edited/replaced' a hidden 4th task called 'git sh' that used cmd.exe to launch bash.exe.

I mad an additional change to the conemu.xml that renamed the default task from cmd to cmder.  this second change was completely unrelated to my msys2_bash change.  

I am going to resubmit to try and clean this up so there will be a total of 5 tasks:

1. cmd
2. powershell
3. powershell as admin
4. git sh - the original 4th task that was hidden
5. git mintty - the task I was trying to add.  It uses mintty instead of cmd to launch bash and seems much faster than task #4.

JanSchulz commented:
If you do this, it would be nice to get a "cmd as admin" task.

BTW: you can push to the branch and it gets refelected in the PR: either push more commits or rebase/squash/commit --amend and force push (`git push -f`).

daxgames commented:
New/Clarified changes:

1. All shells types [ cmd, powershell, bash/mintty] now have working standard user and admin user tasks. (Admin seaions added at request of Jan Schultz above)
2. Shell tasks are grouped by shell types [ cmd, powershell, bash/mintty ] with a cascading menu for each.  I cascaded them to abbreviate and organize the tasks menu.
3. Admin tasks for powershell and bash now launch without file/path not found errors.  When I added the admin tasks they did not work so I fixed them rather than committing broken code.
4. Init.bat launches .\vendor\git-for-windows\post-install.bat if it exists. Note: This file is deleted after it is run.  See vendor\git-for-windows\README.postinstall for details as to why.
5. Bash shell types no longer require init.bat to setup $CMDER_ROOT env variable prior to running /bin/bash, no more exit bash then exit cmd, seems a little faster to launch also.
6. Bash shell types now allow for user level customization via $CMDER_ROOT/config/user-cmder.sh like Cmder and Powershell.

Any questions let me know.

MartiUK commented on packignore:
```diff
@@ -16,3 +16,4 @@ icons\icon_256.png
 Cmder.bat
 vendor\tmp
 appveyor.yml
+vendor\cmder.sh
```
Are you sure this should be ignored?

daxgames replied:
Yes.  The build.ps1 puts it where it needs to be whenrunning a bash shell.  
I ignore it in packing to avoid confusion of having the same file in two 
places, vendor/cmder.sh is not used when running cmder.


On November 10, 2015 3:20:06 AM Martin Kemp <notifications@github.com> wrote:

>> @@ -16,3 +16,4 @@ icons\icon_256.png
>>  Cmder.bat
>>  vendor\tmp
>>  appveyor.yml
>> +vendor\cmder.sh
>
> Are you sure this should be ignored?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cmderdev/cmder/pull/702/files#r44384444

DanielGGordon commented:
Having that 5th task available by default is great. I also renamed the 'cmd' task to 'Cmder' myself. I like the idea of having this as default.

daxgames commented:
Tasks are not exactly as I stated above.  'Cmder' is the default for a reason, see [9427f9a](https://github.com/daxgames/cmder/commit/9427f9a33d71d8c86b8c2e4e881f6e9aa5f3cd43).

The menu looks more like:

```
3. bash
    |_ 4: bash
    |_ 3: bash as Admin
    |_ 2: mintty
    |_ 1: mintty as Admin
2: Powershell
    |_2: Powershell
    |_1: Powershell as Admin
1: cmd
    |_2: Cmder
    |_1: Cmder as Admin
|+|^
```

DanielGGordon commented:
Very nice.

So silly question, but how can I test out the dev branch? Normally I download the zipped file, extract - and it's ready. Do I just put all the files in a folder and that's all I need to do?

MartiUK commented:
@cmderdev/owners @cmderdev/trusted-contributors LGTM :+1:, any thoughts?

Stanzilla commented:
looks ok to me

daxgames commented:
@DanielGGordon since it is not merged into this project yet I guess you could:

1. Pull down my fork into a directory, get it here https://github.com/daxgames/cmder
2. Change to the dir you pulled my fork into and checkout the msys_bash branch.
3. Open powershell.
4. Change to the the scripts dir inside the folder you pulled my fork into.
5. Run '.\build.ps1'.
6. Launch with ''.\cmder.bat"

DanielGGordon commented:
```
PS C:\Users\gordon\workspace\cmder\scripts> .\build.ps1 -verbose
The term '\utils.ps1' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the
spelling of the name, or if a path was included, verify that the path is correct and try again.
At C:\Users\gordon\workspace\cmder\scripts\build.ps1:56 char:2
+ . <<<<  "$PSScriptRoot\utils.ps1"
    + CategoryInfo          : ObjectNotFound: (\utils.ps1:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

The term 'Convertfrom-Json' is not recognized as the name of a cmdlet, function, script file, or operable program. Chec
k the spelling of the name, or if a path was included, verify that the path is correct and try again.
At C:\Users\gordon\workspace\cmder\scripts\build.ps1:60 char:68
+ $sources = Get-Content $sourcesPath | Out-String | Convertfrom-Json <<<<
    + CategoryInfo          : ObjectNotFound: (Convertfrom-Json:String) [], ParentContainsErrorRecordException
```

Is the error message I'm getting. Not sure what's going on here. I cloned your fork, checkout out the correct branch and ran the command you said I should.

daxgames commented:
I just followed the process I outlined andit worked for me.  This is what you should see:

### The initial clone done via Cmder bash:

```
dgames@DTG-SP2 ~
$ git clone  https://github.com/daxgames/cmder.git dax_cmder
Cloning into 'dax_cmder'...
remote: Counting objects: 1458, done.
remote: Compressing objects: 100% (56/56), done.
remote: Total 1458 (delta 30), reused 0 (delta 0), pack-reused 1402
Receiving objects: 100% (1458/1458), 1.55 MiB | 1.92 MiB/s, done.
Resolving deltas: 100% (762/762), done.
Checking connectivity... done.
```

### Then open powershell:

```
C:\Users\Dax
λ cd .\dax_cmder\
C:\Users\Dax\dax_cmder [master ≡]
λ CD .\scripts\
C:\Users\Dax\dax_cmder\launcher [master ≡]
λ git checkout msys_bash
Branch msys_bash set up to track remote branch msys_bash from origin.
Switched to a new branch 'msys_bash'
C:\Users\Dax\dax_cmder\scripts [msys_bash ≡]
λ .\build.ps1 -verbose
VERBOSE: Getting git-for-windows from URL https://github.com/git-for-windows/git/releases/download/v2.6.2.windows.1/PortableGit-2.6.2-32-bit.7z.exe
VERBOSE: Remove tmp/git-for-windows.tmp
VERBOSE: Remove git-for-windows
VERBOSE: Downloading from https://github.com/git-for-windows/git/releases/download/v2.6.2.windows.1/PortableGit-2.6.2-32-bit.7z.exe to
C:\Users\Dax\dax_cmder\vendor\tmp\git-for-windows.tmp
VERBOSE: Getting clink from URL https://github.com/mridgers/clink/releases/download/0.4.5/clink_0.4.5.zip
VERBOSE: Remove tmp/clink.tmp
VERBOSE: Remove clink
VERBOSE: Downloading from https://github.com/mridgers/clink/releases/download/0.4.5/clink_0.4.5.zip to C:\Users\Dax\dax_cmder\vendor\tmp\clink.tmp
VERBOSE: Getting conemu-maximus5 from URL https://github.com/Maximus5/ConEmu/releases/download/v15.10.25/ConEmuPack.151025.7z
VERBOSE: Remove tmp/conemu-maximus5.tmp
VERBOSE: Remove conemu-maximus5
VERBOSE: Downloading from https://github.com/Maximus5/ConEmu/releases/download/v15.10.25/ConEmuPack.151025.7z to C:\Users\Dax\dax_cmder\vendor\tmp\conemu-maximus5.tmp
VERBOSE: Getting clink-completions from URL https://github.com/vladimir-kotikov/clink-completions/archive/0.2.1.zip
VERBOSE: Remove tmp/clink-completions.tmp
VERBOSE: Remove clink-completions
VERBOSE: Downloading from https://github.com/vladimir-kotikov/clink-completions/archive/0.2.1.zip to C:\Users\Dax\dax_cmder\vendor\tmp\clink-completions.tmp
WARNING: You are not building a launcher, Use -Compile
WARNING: This cannot be a release. Test build only!
VERBOSE: Adding cmder.sh /etc/profile.d
VERBOSE: All good and done!
```

And yes it is the same basic process to test cmder's 'development' branch.  The difference is where you clone the initial repo from and which branch you checkout.

DanielGGordon commented:
Hmm dunno what to tell you. Cloned, checkout the branch, went into the scripts directory, and tried the command - getting that error.

Jackbennett commented:
It looks like your `$PSScriptRoot` variable is empty what version of powershell are you on with `$PSVersionTable`

DanielGGordon commented:
```
PS C:\Users\gordon> echo $PSVersionTable

Name                           Value
----                           -----
CLRVersion                     2.0.50727.5485
BuildVersion                   6.1.7601.17514
PSVersion                      2.0
WSManStackVersion              2.0
PSCompatibleVersions           {1.0, 2.0}
SerializationVersion           1.1.0.1
PSRemotingProtocolVersion      2.1
```

Jackbennett commented:
Yeh PS version 2. I think we found $PSscriptroot a 3.0 thing. either just replace it with a dot or install the new window management framework 4 or 5. 

I think I'll change it to a dot anyway since it's already specific to is execution path. it's fancier than it needs to be. I've only got a phone I can't use an editor till Monday,  who am @martyuk ?

daxgames commented:
Maybe we should change to the below that dows work on Powershell 2:

### Script:
```
Write-Host ("MyCommand.Path from function: " + $MyInvocation.MyCommand.Path)
write-host ("MyCommand.DirectoryName: " + $MyInvocation.MyCommand.Name)
write-host ("MyCommand.DirectoryName: " + ($MyInvocation.MyCommand.Path).tostring().replace(("\" + $MyInvocation.MyCommand.Name), $null) )
```

### Output: 
```
dgames@DTG-SP2 dgames.DTG> $PSVersionTable

Name                           Value
----                           -----
CLRVersion                     2.0.50727.8669
BuildVersion                   6.1.7600.16385
PSVersion                      2.0
WSManStackVersion              2.0
PSCompatibleVersions           {1.0, 2.0}
SerializationVersion           1.1.0.1
PSRemotingProtocolVersion      2.1


dgames@DTG-SP2 dgames.DTG> .\Documents\test.ps1
MyCommand.Path from function: C:\Users\dgames.DTG\Documents\test.ps1
MyCommand.Name: test.ps1
MyCommand.DirectoryName: C:\Users\dgames.DTG\Documents
```

daxgames commented:
synced with upstream and squashed

MartiUK commented:
Would you be able to write some documentation up on your changes @daxgames? For new users.

DanielGGordon commented:
"This branch is up-to-date with the base branch" 

What is the 'base branch'? Is that development in this case or master? It would be nice if github specified here what branch that is. A bit confused on what this means. Sorry for noob question.

MartiUK commented:
@DanielGGordon run `git status` it should tell you what branch you're on. If you've checked out @daxgames's fork then you should run `git checkout msys-bash`.

DanielGGordon commented:
Ok I was looking through the git history and I got my answer. I can see that the development branch was pulled into the `msys_bash` branch - that's all I wanted to know. I could figure it out from the `git log` but I couldn't tell just by looking at this page.

daxgames commented:
I could. Where would you like me to put it?  Just wanting to be sure how it 
will be made available so I know what to write?  I have not seen much 
documentation.


On November 19, 2015 7:32:30 AM Martin Kemp <notifications@github.com> wrote:

> Would you be able to write some documentation up on your changes @daxgames? 
> For new users.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cmderdev/cmder/pull/702#issuecomment-158057557

MartiUK commented:
@daxgames Any instructions needed in the readme. Any changes to key bindings I can put on the website.

DanielGGordon commented:
Ok sweet I got it compile and running, although a few things I've noticed. 

I noticed the vim issues are gone (sweet!)

1. The mintty task has a black background, whereas others are more gray. 

~~My `/etc/profile.d/git-prompt.sh` no longer is executed - or at least the prompt that I've created in there does not work. I get the standard `user@hostname` prompt. Even if I try to do:
`source /etc/profile.d/git-prompt.sh`, I don't see a change in my prompt.~~

Nevermind that, I realized that the mintty task uses its own git home directory, therefor pointing to a different `git-prompt.sh`.

2. ctrl + v doesn't work for pasting into the mintty task. Normally with git bash, you can use `ctrl + insert` to copy, and `shift + insert` to paste.

3. The cmder icon is gone from the taskbar and the top of the window too. All I see is the conemu window. But this could have to do with the fact that I already had Cmder installed elsewhere.

daxgames commented:
Hey Martin, I don't think there are any of bits like you suggested that 
need to be documented as long as the Git for Windows postinstall.bat file 
is included in the packed distribution everything should be automatic. 
There were no key binding changes that I'm aware of, if so  they were 
accidental.

Main things I added to the bash support were the ability use the native Git 
for Windows terminal 'mintty', to do cmder portable user-specific config 
from the config/user-cmder.sh file and that the /etc/profile.d/cmder.sh 
will source the %userprofile%\.bashrc if it exists.

Biggest default user visible changes were organisation of the tasks menu 
into a cascading menu by shell type.



On November 20, 2015 4:36:01 AM Martin Kemp <notifications@github.com> wrote:

> @daxgames Any instructions needed in the readme. Any changes to key 
> bindings I can put on the website.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cmderdev/cmder/pull/702#issuecomment-158351756

daxgames commented:
@DanielGGordon because mintty.exe is a true terminal and does not use the windows console api what you are seeing is probably normal for 'mintty'.  Simply hi-lighting copies and right click at the command line pastes.

Not sure what you mean by 'I realized that the mintty task uses its own git home directory, therefor pointing to a different  git-prompt.sh'

daxgames commented:
@DanielGGordon If you want to add the solarized dark palette to minty for a more pleasing scheme do the following for cmder to make it portable:

NOTE: This only works if you are running my msys_bash branch, at least until it gets merged into the cmder project.

From a bash/mintty shell:

```
cd $CMDER_ROOT/vendor
git clone https://github.com/karlin/mintty-colors-solarized.git
cd mintty-colors-solarized/
echo source \$CMDER_ROOT/vendor/mintty-colors-solarized/mintty-solarized-dark.sh>>$CMDER_ROOT/config/user-cmder.sh
```

See: https://github.com/karlin/mintty-colors-solarized
For details on Solarized and why to use it: http://ethanschoonover.com/solarized
If in Mintty close and restart minty to see the new color scheme.

DanielGGordon commented:
> Not sure what you mean by 'I realized that the mintty task uses its own git home directory, therefor pointing to a different git-prompt.sh'

Yea don't worry about that - I figured it out. I already have git bash, and was wondering why my git-prompty.sh script wasn't being loaded - and that's because it's outside of the Cmder build. 

As for the solarized theme - that seems to look nice. Good find. I will have to tweak it a little to my liking. I'll have to play around with `%USERPROFILE%.mintyrc`.

I'm also a bit confused on what is going on here, sorry if this is out of the scope of this pull request.

I am confused about mintty vs. git bash. What is the difference? To be honest, I don't know what I've been using this whole time (outside of Cmder) - actually I just looked, and I guess I've been using Mintty. So what is the difference between using that, and the `git bash` task? 

When I go to my GIT folder in explorer, I see the options `GIT Bash` and GIT CMD`. The context menu says "GIT Bash here" - is that mintty or not?

Lastly, the `Bash:: bash as admin` and `cmd::Cmder as Admin` (and probably non-admin too) task still results in the strange vim issues (everything highlighted in gray, doesn't exit properly).

daxgames commented:
@DanielGGordon the difference between the bash::mintty and bash::bash tasks are the terminal/console used to interract with the system.  I have eluded to this before.  mintty is a true graphical terminal loading bash where the other is basically the windows cmd.exe loading bash using the windows console api.  I believe ConEmu, the app behind cmder, was designed primarily to work with the windows console api but it can also integrate basic terminal apps like mintty and putty with some differences in functionality.

As far as git bash vs. git cmd in your folder structure the first one uses the bash.exe(unix-ish) one uses cmd.exe(windows).  In older git for windows it is the similar to bash::bash and cmd::cmder both using the Windows console API.  In newer git for windows, what is in the development branch, it is the similar to bash::mintty and cmd::cmder.

If you have been using mintty outside of cmder the issues you are having with vim and grey hi-lighting could be related to running vim under the windows console api and an incompatible vim colorscheme.  Cmd.exe is only capable of displaying 16 colors while mintty can display many more.  I see the same issues with vim in cmder not using mintty if I try to use a colorscheme like solarized.vim.  What colorscheme are you running with vim?  Try opening a standard Windows cmd.exe prompt and running vim using a fully qualified path to vim.exe.  I am guessing you will see the same issue.

DanielGGordon commented:
> Try opening a standard Windows cmd.exe prompt and running vim using a fully qualified path to vim.exe. I am guessing you will see the same issue.

So I opened `cmd` (no Cmder), and opened a file using the vim that came with your build (this vim executable is located inside the Cmder folder I created for your branch).

When I did that, I did get the same issue.

However, when I open the same file, using the vim from my own git installation in `C:/Program Files/Git/usr/bin/`, then I do not see the issue.

My `.vimrc` in my home directory (which is used by both vim executables) does not have a theme or font specified. I am pretty sure both vims use the same `.vimrc` file. Is there any way the vim colorscheme is changed besides from inside the `.vimrc` file?

Also, I mentioned earlier about the Conemu icon showing up. Also the tasks don't have icons (not sure if this is intentional).

![image](https://cloud.githubusercontent.com/assets/11824880/11341398/f05ff524-91c7-11e5-9883-c26775c97f08.png)

![image](https://cloud.githubusercontent.com/assets/11824880/11341392/ea1deedc-91c7-11e5-96f3-4ecae9263950.png)

and the top bar of Cmder:

![image](https://cloud.githubusercontent.com/assets/11824880/11341418/00df2cda-91c8-11e5-953c-f6d38f851dc1.png)

daxgames commented:
@DanielGGordon - The icons not showing are because when you build the development branch you are not building cmder.exe which is where the icons come from in the task definition.  Has nothing to do with my branch just that you are not building that EXE, probably because you do not have the necessary tools to compile and it may even require a command line option.  I have not looked.

Your .vimrc may be setting a colorscheme or including a vim file that is setting a colorscheme.  If I remove/comment all 'colorscheme ...' lines from my .vimrc it all works for me in all shells and issuing ':colorscheme' in vim returns 'default'.  So nothing in the cmder development branch is changing the vim colorscheme from default.  What do you get when you issue ':colorscheme' in vim?

DanielGGordon commented:
Issued `:colorscheme` in vim, and it returned `default`

daxgames commented:
I don't know.  The only variable I can see is the %userprofile%\.vimrc and 
any config files it may include.  If you are getting Grey vim 
polite/background with color scheme of default when using Windows console 
api I don't have an answer for you other than I am not seeing the same 
using the same checked out branch you are.  My branch does not modify 
anything with the vim config.

What if you launch vim from cmd.exe from your old cmder folder outside 
cmder?  Same behavior?

Can you post your .vimrc?   Is 's 'set t_Co=256' in your .vimrc?  I get 
some grey weirdness if it is an not using mintty.

Do you by chance have a _vimrc?





On November 23, 2015 2:57:26 PM Daniel Gordon <notifications@github.com> wrote:

> Issued `:colorscheme` in vim, and it returned `default`
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cmderdev/cmder/pull/702#issuecomment-159061761

DanielGGordon commented:
set term=xterm
    set t_Co=256
    let &t_AB="\e[48;5;%dm"
    let &t_AF="\e[38;5;%dm"
    syntax on
    set tabstop=8 softtabstop=0 expandtab shiftwidth=4 smarttab

`$TERM` in bash task in Cmder - cygwin
`%TERM%` in Cmder task - cygwin

I don't have a `_vimrc` file.

> What if you launch vim from cmd.exe from your old cmder folder outside
cmder?  Same behavior?

I actually don't have a version of Cmder installed where this is not an issue. 

Is there some other place besides `~/.vimrc` that my `.vimrc` file could be found?

Also I removed the `set t_Co=256` line from the `vimrc`, and I still have the same problem.

daxgames commented:
Remove term=xterm from .vimrc.


On November 23, 2015 4:12:15 PM Daniel Gordon <notifications@github.com> wrote:

>     set term=xterm
>     set t_Co=256
>     let &t_AB="\e[48;5;%dm"
>     let &t_AF="\e[38;5;%dm"
>     syntax on
>     set tabstop=8 softtabstop=0 expandtab shiftwidth=4 smarttab
>
> `$TERM` in bash task in Cmder - cygwin
> `%TERM%` in Cmder task - cygwin
>
> I don't have a `_vimrc` file.
>
>> What if you launch vim from cmd.exe from your old cmder folder outside
> cmder?  Same behavior?
>
> I actually don't have a version of Cmder installed where this is not an issue.
>
> Is there some other place besides `~/.vimrc` that my `.vimrc` file could be 
> found?
>
> Also I removed the `set t_Co=256` line from the `vimrc`, and I still have 
> the same problem.
>
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cmderdev/cmder/pull/702#issuecomment-159082529

DanielGGordon commented:
Ah it was the

    let &t_AB="\e[48;5;%dm"
    let &t_AF="\e[38;5;%dm"

stuff in the file as well. Thanks for all your held @daxgames! It looks good now! Very happy to be using Cmder again (and with solarized!)

daxgames commented:
Glad to help.


On November 23, 2015 4:38:12 PM Daniel Gordon <notifications@github.com> wrote:

> Ah it was the
>
>     let &t_AB="\e[48;5;%dm"
>     let &t_AF="\e[38;5;%dm"
>
> stuff in the file as well. Thanks for all your held @daxgames! It looks 
> good now! Very happy to be using Cmder again (and with solarized!)
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/cmderdev/cmder/pull/702#issuecomment-159089948

daxgames commented:
@MartiUK - Checkout the docs I added to the README.md.  I think that about covers what I did.  Let me know if you need more/different.

MartiUK commented:
Perfect, thanks @daxgames

Stanzilla commented:
One thing I noticed when using bash is that after I ctrl+c a command, the next stuff I write is not shown. I can hit enter and it works but the text is not showing up. Had anything like that happen @daxgames?
